packages/kbn-lens-embeddable-utils/config_builder/utils.ts

```diff
@@ -226,7 +223,14 @@ export const buildDatasourceStates = async (
         getValueColumns
       );
       if (layerConfig) {
-        layers[type]!.layers[layerId] = layerConfig;
+        layers = {
+          ...layers,
+          [type]: {
+            layers: {
+              [layerId]: layerConfig,
+            },
+          },
+        };
```

Comment:
ℹ️ I changed the implementation here to return the layers with only the active datasource because it was causing problems due to this funtion

```
function getActiveDatasourceIdFromDoc(doc?: Document) {
  if (!doc) {
    return null;
  }

  const [firstDatasourceFromDoc] = Object.keys(doc.state.datasourceStates);
  return firstDatasourceFromDoc || null;
}
```

This is used to detect the activeDatasource. As you see it returns the first item from the datasources array. If there are 2, it doesnt guarantee that the correct datasourceId will be returned. I think we should refactor it and check if the datasources in the state have layers. But for now I fixed that on our builder